### PR TITLE
feat: Adding Support for Local Tap

### PR DESCRIPTION
**Pull request by CoshUS**
*Issue #, if available:*
N/A
*Description of changes:*
Adding support for building bottles with local files instead of remote tap.
Files of this repo will be copied into homebrew's tap directory and tricking it for bottle building.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: 1.0.{build}
 branches:
   # whitelist
   only:
-    - master
+    - sam-cli-release
```
Comment (mndeveci) [inline appveyor.yml:6]: Are we going to keep this new branch here, or this was used just for testing?

File: build-formula.sh [hunk 2]
```diff
@@ -13,6 +13,13 @@ export HOMEBREW_NO_INSTALL_CLEANUP=1
 # Commenting out the auto upgrade since appveyor mac has an old version and fails buildilng bottles for SAM CLI
 # export HOMEBREW_NO_AUTO_UPDATE=1
 
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    test -d ~/.linuxbrew && eval "$(~/.linuxbrew/bin/brew shellenv)"
+    test -d /home/linuxbrew/.linuxbrew && eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+    test -r ~/.bash_profile && echo "eval \"\$($(brew --prefix)/bin/brew shellenv)\"" >>~/.bash_profile
+    echo "eval \"\$($(brew --prefix)/bin/brew shellenv)\"" >>~/.profile
+fi
```
Comment (mndeveci) [inline build-formula.sh:21]: I assume we are trying to setup brew shell configuration here. Can we put some comments on top about what this piece is trying to achieve?

File: update-local-tap.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+TAP_NAME="aws/homebrew-tap"
```
Comment (mndeveci) [inline update-local-tap.sh:4]: Curious, isn't tap name is `aws/tap`? Is it different when running locally?

File: appveyor.yml (hunk 1 above)
Comment (CoshUS) [reply appveyor.yml:6]: This branch will be the one we are making changes to during release. We will merge it to master at the end of release.

File: build-formula.sh (hunk 2 above)
Comment (CoshUS) [reply build-formula.sh]: This was needed for local running in some cases. Removed as this is not needed for AppVeyor.

File: update-local-tap.sh (hunk 3 above)
Comment (CoshUS) [reply update-local-tap.sh]: Homebrew will automatically add `homebrew-` in front of the name in the command `brew tap` but the folder and location on disk is still the full repo name. 
Added comments to explain this behavior.

**Review by qingchm — APPROVED** [review]

**Review by mndeveci — APPROVED** [review]

**Pull request merged**